src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+            return
+
+        self._append_primary_secondary_angles(parts, col_idx, primary, secondary)
+        self._append_primary_hit_geometry(
+            parts, col_idx, radius_m, primary_hit_x_m, primary_hit_y_m
+        )
+        self._append_secondary_hit_geometry(
+            parts, col_idx, secondary_radius_m, secondary_hit_x_m, secondary_hit_y_m
+        )
+
+    def _append_primary_secondary_angles(self, parts, col_idx, primary, secondary):
+        """Append primary/secondary angle values (or NaN) if arrays are provided.
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Indices for angle columns.
+        primary, secondary : list | None
+            Output arrays to append into.
+        """
+        if primary is not None:
+            primary.append(self._parse_float_with_nan(parts, col_idx.get("primary")))
+        if secondary is not None:
+            secondary.append(self._parse_float_with_nan(parts, col_idx.get("secondary")))
+
+    def _append_primary_hit_geometry(
+        self, parts, col_idx, radius_m, primary_hit_x_m, primary_hit_y_m
+    ):
+        """Append primary-mirror hit geometry (radius and x/y in meters).
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Indices for hit coordinate columns.
+        radius_m, primary_hit_x_m, primary_hit_y_m : list | None
+            Output arrays to append into.
+        """
+        x_ok, x_cm = self._parse_float(parts, col_idx.get("prim_x"))
+        y_ok, y_cm = self._parse_float(parts, col_idx.get("prim_y"))
+        if x_ok and y_ok:
+            r_m = ((x_cm**2 + y_cm**2) ** 0.5) / 100.0
+            if radius_m is not None:
+                radius_m.append(r_m)
+            if primary_hit_x_m is not None:
+                primary_hit_x_m.append(x_cm / 100.0)
+            if primary_hit_y_m is not None:
+                primary_hit_y_m.append(y_cm / 100.0)
+            return
+        if radius_m is not None:
+            radius_m.append(float("nan"))
+        if primary_hit_x_m is not None:
+            primary_hit_x_m.append(float("nan"))
+        if primary_hit_y_m is not None:
+            primary_hit_y_m.append(float("nan"))
+
+    def _append_secondary_hit_geometry(
+        self, parts, col_idx, secondary_radius_m, secondary_hit_x_m, secondary_hit_y_m
+    ):
+        """Append secondary-mirror hit geometry (radius and x/y in meters).
+
+        Parameters
+        ----------
+        parts : list[str]
+            Tokenized input row.
+        col_idx : dict[str, int]
+            Indices for hit coordinate columns.
+        secondary_radius_m, secondary_hit_x_m, secondary_hit_y_m : list | None
+            Output arrays to append into.
+        """
+        sx_ok, sx_cm = self._parse_float(parts, col_idx.get("sec_x"))
```

Comment:
see previous comment. Simplify for readability.

Reply:
fixed